Kernel/Devices/Storage/USB/MassStorageDevice.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023, Leon Albrecht <leon.a@serenityos.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/IntrusiveRedBlackTree.h>
+#include <Kernel/Bus/USB/USBController.h>
+#include <Kernel/Bus/USB/USBDevice.h>
+#include <Kernel/Devices/Storage/StorageDevice.h>
+#include <Kernel/Devices/Storage/USB/BulkSCSIInterface.h>
+
+namespace Kernel::USB {
+
+class MassStorageDevice : public USB::Device {
+public:
+    static ErrorOr<NonnullLockRefPtr<MassStorageDevice>> try_create_from_device(USB::Device const& base_device);
```

Comment:
This doesn't make sense in my opinion. we already have a USB device struct, no need to create a more specialized one. See my suggestions on how to use @Quaker762 code and my code in #21085.